pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -716,17 +688,24 @@ def make_xy_term(q1: QubitId, q2: QubitId) -> qutip.Qobj:
             """
             dist = np.linalg.norm(self._qdict[q1] - self._qdict[q2])
             coords_dim = len(self._qdict[q1])
-            mag_norm = np.linalg.norm(self._seq.magnetic_field[:coords_dim])
+            mag_norm = np.linalg.norm(
+                cast(
+                    np.ndarray,
+                    self.samples_obj._magnetic_field,
+                )[:coords_dim]
```

Comment:
Since you are casting again below, maybe cast once and put it in a variable instead